src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1033,6 +1033,18 @@ static StageInfo getStageInfo(JsonNode jsonNode) throws SnowflakeSQLException {
       }
     }
 
+    boolean useS3RegionalUrl;
+    if (stageInfo.getStageType() == StageInfo.StageType.S3) {
+      // This node's value is set if PUT is used without Session. (For Snowpipe Streaming, we rely
+      // on a response from a server to have this field set to use S3RegionalURL)
+      JsonNode useS3RegionalURLNode =
+          jsonNode.path("data").path("stageInfo").path("useS3RegionalUrl");
+      if (!useS3RegionalURLNode.isMissingNode()) {
```

Comment:
Do you like to check for useS3RegionalURLNode null or not. What will happen "useS3RegionalUrl" path doesn't exist?

Reply:
if the path doesnt exist, we will not change stageInfo

Reply:
ok